packages/video_player/video_player_platform_interface/lib/messages.dart

```diff
@@ -0,0 +1,271 @@
+// See also: https://pub.dev/packages/pigeon
+import 'package:flutter/services.dart';
+
+class VoidMessage {
+  // ignore: unused_element
+  Map _toMap() {
+    Map dartleMap = Map();
+    return dartleMap;
+  }
+  // ignore: unused_element
+  static VoidMessage _fromMap(Map dartleMap) {
+    var result = VoidMessage();
+    return result;
+  }
+}
+
+class TextureMessage {
+  int textureId;
+  // ignore: unused_element
+  Map _toMap() {
+    Map dartleMap = Map();
+    dartleMap["textureId"] = textureId;
+    return dartleMap;
+  }
+  // ignore: unused_element
+  static TextureMessage _fromMap(Map dartleMap) {
+    var result = TextureMessage();
+    result.textureId = dartleMap["textureId"];
+    return result;
+  }
+}
+
+class CreateMessage {
+  String asset;
+  String uri;
+  String package;
+  String formatHint;
+  // ignore: unused_element
+  Map _toMap() {
+    Map dartleMap = Map();
+    dartleMap["asset"] = asset;
+    dartleMap["uri"] = uri;
+    dartleMap["package"] = package;
+    dartleMap["formatHint"] = formatHint;
+    return dartleMap;
+  }
+  // ignore: unused_element
+  static CreateMessage _fromMap(Map dartleMap) {
+    var result = CreateMessage();
+    result.asset = dartleMap["asset"];
+    result.uri = dartleMap["uri"];
+    result.package = dartleMap["package"];
+    result.formatHint = dartleMap["formatHint"];
+    return result;
+  }
+}
+
+class LoopingMessage {
+  int textureId;
+  bool isLooping;
+  // ignore: unused_element
+  Map _toMap() {
+    Map dartleMap = Map();
+    dartleMap["textureId"] = textureId;
+    dartleMap["isLooping"] = isLooping;
+    return dartleMap;
+  }
+  // ignore: unused_element
+  static LoopingMessage _fromMap(Map dartleMap) {
+    var result = LoopingMessage();
+    result.textureId = dartleMap["textureId"];
+    result.isLooping = dartleMap["isLooping"];
+    return result;
+  }
+}
+
+class VolumeMessage {
+  int textureId;
+  double volume;
+  // ignore: unused_element
+  Map _toMap() {
+    Map dartleMap = Map();
+    dartleMap["textureId"] = textureId;
+    dartleMap["volume"] = volume;
+    return dartleMap;
+  }
+  // ignore: unused_element
+  static VolumeMessage _fromMap(Map dartleMap) {
+    var result = VolumeMessage();
+    result.textureId = dartleMap["textureId"];
+    result.volume = dartleMap["volume"];
+    return result;
+  }
+}
+
+class PositionMessage {
+  int textureId;
+  int position;
+  // ignore: unused_element
+  Map _toMap() {
+    Map dartleMap = Map();
+    dartleMap["textureId"] = textureId;
+    dartleMap["position"] = position;
+    return dartleMap;
+  }
+  // ignore: unused_element
+  static PositionMessage _fromMap(Map dartleMap) {
+    var result = PositionMessage();
+    result.textureId = dartleMap["textureId"];
+    result.position = dartleMap["position"];
+    return result;
+  }
+}
+
+class VideoPlayerApi {
+  Future<VoidMessage> initialize(VoidMessage arg) async {
+    Map requestMap = arg._toMap();
+    BasicMessageChannel channel =
```

Comment:
I guess this part is a bit different from the receiving side. I get the ergonomics of it but I wonder what's the cost (if a function is invoked on each frame). Is it reasonable to save it somewhere?